docs/book/modules.md

```diff
@@ -0,0 +1,105 @@
+# Modules
+
+A module is a PHP namespace. It can contain controllers, classes, view scripts, configuration, tests, as well as public assets such as images, CSS, and JavaScript. For example, if you have an e-commerce application, you could have a module for presenting and selling products, and another for administrators to create products and manage orders.
+
+## Structure
+
+Given a module named `Application`, here is the recommended structure:
+
+```bash
+Application/
+    config/
+        module.config.php
+    public/
+        images/
+        css/
+        js/
+    src/
+        Controller/
+            ProductController.php
+        Module.php
+        <other classes>
+    test/
+        Controller/
+            ProductControllerTest.php
+    view/
+        application/
+            product/
+                list.phtml
+        layout/
+            layout.phtml
+```
+
+## Loading
+
+All module classes can be autoloaded by adding them as a namespace in composer.json:
+
+```json
+"autoload": {
+    "psr-4": {
+        "Application\\": "module/Application/src/"
+    }
+}
+```
+
+This means that whenever you refer to a class in the `Application` namespace, PHP will look for it in the `module/Application/src/` folder. A class named `Application\Controller\ProductController` would be found in the file `module/Application/src/Controller/ProductController.php`.
+
+In addition to autoloading, the application needs to be aware that the `Application` namespace represents a module. This is accomplished by adding it to `config/modules.config.php`:
+
+```php
+return [
+    'Application',
+    // other modules
+];
+```
+
+## Configuration
+
+// ...
+
+## Listeners
+
+Each module has a `Module` class, which can contain a variety of methods called listeners. These listeners are called by Laminas MVC to configure the module.
+
+For the namespace `Application`, Laminas MVC will look for `Application\Module`. Here is a simple example that loads a module-specific configuration file:
+
+```php
+namespace Application;
+
+class Module
+{
+    public function getConfig(): array
+    {
+        /** @var array $config */
+        $config = include __DIR__ . '/../config/module.config.php';
+        return $config;
+    }
+}
+```
+
+### getModuleDependencies()
+
+Checks each module to verify whether all the modules it depends on were
+loaded. Each of the values returned by the method is checked
+against the loaded modules list: if one of the values is not in that list, a
+`Laminas\ModuleManager\Exception\MissingDependencyModuleException` is thrown.
+
+```php
+namespace Application;
+
+use Laminas\ModuleManager\Feature\DependencyIndicatorInterface;
```

Comment:
An exception class and an interface of laminas-modulemanager are presented here, but the component itself is not mentioned before.

Reply:
From the developer's perspective, there is no value in knowing that this pertains to this component. We can reference it in Learn More, but I don't see the value in explaining this component here. In fact, I find that this is what made the previous docs confusing.

Reply:
> …but I don't see the value in explaining this component here.

The component should not be explained, but at least mentioned. Otherwise it is confusing that another namespace plays a role now.

Reply:
Ah, I see what you mean. Will do.

Reply:
Thank you and sorry for the misleading.

Reply:
It turns out that I referred to Laminas MVC when I should have referred to Laminas Module Manager. This makes it easier to mention the component and link to it.

By the way, do we prefer to spell out as `Laminas Module Manager`, or use the package name `laminas-modulemanager`? I have occasionally seen component names spelled out, so am not sure what is the preferred approach. I personally prefer to spell out, but I can follow any guidelines.

Reply:
For some reason, I have a commit that shows as the totally wrong date on GitHub. Here is my fix for this comment: https://github.com/laminas/laminas-mvc/pull/103/commits/c4dad7b155028b412e08b5b9e78a8504bde22f8a

Reply:
The package name is used most and is also the preferred way so far.

Reply:
If something refers to Mezzio, then the spelling with a capital initial is used: "Mezzio". The package names also follow the names used in Composer. Example: "mezzio-problem-details".